library/std/src/sys/pal/common/exit_guard.rs

```diff
@@ -0,0 +1,59 @@
+cfg_if::cfg_if! {
+    if #[cfg(target_os = "linux")] {
+        /// Mitigation for <https://github.com/rust-lang/rust/issues/126600>
+        ///
+        /// On UNIX-like platforms (where `libc::exit` may not be thread-safe), ensure that only one
+        /// Rust thread calls `libc::exit` (or returns from `main`) by calling this function before
+        /// calling `libc::exit` (or returning from `main`).
+        ///
+        /// Technically not enough to ensure soundness, since other code directly calling
+        /// libc::exit will still race with this.
```

Comment:
```suggestion
        /// On glibc, `libc::exit` has been observed to not always be thread-safe.
        /// It is currently unclear whether that is a glibc bug or allowed by the standard.
        /// To mitigate this problem, we ensure that only one
        /// Rust thread calls `libc::exit` (or returns from `main`) by calling this function before
        /// calling `libc::exit` (or returning from `main`).
        ///
        /// Technically, this is not enough to ensure soundness, since other code directly calling
        /// `libc::exit` will still race with this.
```

Reply:
I'd add the issue number #126600 as well.

Reply:
The issue is linked above this on line 3, is that good?